mypy/build.py

```diff
@@ -2303,6 +2303,7 @@ def find_module_and_diagnose(manager: BuildManager,
         if (root_source  # Honor top-level modules
                 or (not path.endswith('.py')  # Stubs are always normal
                     and not options.follow_imports_for_stubs)  # except when they aren't
+                or id == 'typing'  # typing is not a stub, and is always normal
```

Comment:
I think this makes sense, to make it more general may be use `id not in mypy.semanal_main.core_modules` instead of this line and the below one?